sycl/source/detail/global_handler.cpp

```diff
@@ -71,6 +71,49 @@ std::atomic_uint ObjectUsageCounter::MCounter{0};
 GlobalHandler::GlobalHandler() = default;
 GlobalHandler::~GlobalHandler() = default;
 
+void GlobalHandler::InitXPTIStuff() {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  // Let subscribers know a new stream is being initialized
+  getXPTIRegistry().initializeStream(SYCL_STREAM_NAME, GMajVer, GMinVer,
+                                     GVerStr);
+  xpti::payload_t SYCLPayload("SYCL Runtime Exceptions");
+  uint64_t SYCLInstanceNo;
+  GSYCLCallEvent = xptiMakeEvent("SYCL Try-catch Exceptions", &SYCLPayload,
+                                 xpti::trace_algorithm_event, xpti_at::active,
+                                 &SYCLInstanceNo);
+#endif
+}
+
+void GlobalHandler::TraceEventXPTI(const char *Message) {
+  if (!Message)
+    return;
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  if (xptiTraceEnabled()) {
+    // We have to handle the cases where: (1) we may have just the code location
+    // set and not UID and (2) UID set
+    detail::tls_code_loc_t Tls;
+    auto CodeLocation = Tls.query();
+
+    // Creating a tracepoint will convert a CodeLocation to UID, if not set
```

Comment:
Hi @tovinkere I would suggest to left responsibility how to treat absence of data to subscribers. Adding extra strings to this data brings more contracts between rt and xpti subscribers which I think could lead to confusion and necessity to align them each time we make any change.
If you will take a look at unit tests subscriber implementation - it handles it by its own and decides what to print. The same is for sycl-trace subscriber.

Reply:
If we are sending in invalid event types on occasion where code location is not available, we should be documenting it or the "sycl" stream as one of the possibilities for the trace type in question so the callback handler can handle it. All other trace types that are a part of this stream will be sending in valid events, if I am not mistaken. Either we fix it in code or we add documentation.

Reply:
@tovinkere why is it invalid event? If we not specify code location - event should be still valid. We still could trace exception using its message. When subscriber receives event it will check if it has code location since we have flags to check. Now we could not guarantee that any exception sent will contain code location - at least because not all APIs has instrumentation. But it is not the reason to not trace exceptions at all.

Reply:
@KseniyaTikhomirova The section in the [documentation ](https://github.com/intel/llvm/blob/sycl/sycl/doc/design/SYCLInstrumentationUsingXPTI.md) for the "sycl" stream should be updated to include expected inputs for xpti::trace_diagnostics.

Reply:
Thanks for pointing me to that, updated https://github.com/intel/llvm/pull/8101/commits/be352323eee8780bc99fbc7bfee562919225578f 😊